src/machine/machine_nrf.go

```diff
@@ -185,21 +185,21 @@ func (i2c I2C) Tx(addr uint16, w, r []byte) error {
 		}
 	}
 	if len(r) != 0 {
+		// To trigger suspend task when a byte is received
+		i2c.Bus.SHORTS &^= (nrf.TWI_SHORTS_BB_STOP << nrf.TWI_SHORTS_BB_STOP_Pos)
```

Comment:
I'm sorry that I made a mistake before. The correct syntax here would be better stated `i2c.Bus.SHORTS &^= nrf.TWI_SHORTS_BB_STOP` like you have below.

Reply:
I'm not sure I understand - we want i2c.Bus.SHORTS to be set to 1 here, don't we? &^= nrf.TWI_SHORTS_BB_STOP isn't going to do that is it?

Reply:
In that case you want `i2c.Bus.SHORTS = nrf.TWI_SHORTS_BB_STOP` as the `&^=` operator turns the bit OFF.

Reply:
How about like this with TWI_SHORTS_BB_SUSPEND?

Reply:
Now I am confused, sorry. What is it exactly you want to set or turn off here?

Reply:
I think the idea is to:
- at the start of a read set SHORTS to 1, which is TWI_SHORTS_BB_SUSPEND
- before the last byte of the read is read set SHORTS to 2, which is TWI_SHORTS_BB_STOP
- when the read is finished set SHORTS to 0, which is &^= nrf.TWI_SHORTS_BB_STOP? Or perhaps that should be TWI_SHORTS_BB_STOP_Disabled?

Reply:
Seems like that would be best expressed as:

```
i2c.Bus.SHORTS = sam.TWI_SHORTS_BB_SUSPEND
```

```
i2c.Bus.SHORTS = sam.TWI_SHORTS_BB_STOP
```

```
i2c.Bus.SHORTS = sam.TWI_SHORTS_BB_SUSPEND_Disabled
```